lib/munster/jobs/processing_job.rb

```diff
@@ -5,14 +5,23 @@
 module Munster
   class ProcessingJob < ActiveJob::Base
     def perform(webhook)
+      webhook.class.transaction do
+        webhook.with_lock do
+          return unless webhook.received?
+          webhook.processing!
+        end
+      end
+
       if webhook.handler.valid?(webhook.request)
-        # TODO: we are going to add some default state lifecycle managed
-        # by the background job later
         webhook.handler.process(webhook)
+        webhook.processed! if webhook.processing?
       else
-        Rails.logger.info { "Webhook #{webhook.inspect} did not pass validation and was skipped" }
+        Rails.logger.info { "#{webhook.class} #{webhook.id} did not pass validation and was skipped" }
```

Comment:
**issue:** I would consider this to be an error-level severity and would like to see it in AppSignal, instead of just as a message in the logs (which I have a 99% chance of missing).

Reply:
Fair enough